website/common/script/ops/buy/buyQuest.js

```diff
@@ -10,7 +10,9 @@ import get from 'lodash/get';
 // buy a quest with gold
 module.exports = function buyQuest (user, req = {}, analytics) {
   let key = get(req, 'params.key');
-  let quantity = req.quantity || 1;
+
+  if (req.quantity && isNaN(req.quantity)) throw new BadRequest(i18n.t('invalidQuantity', req.language));
+  let quantity = req.quantity ? Number(req.quantity) : 1;
```

Comment:
both here and in the other file, can you convert to Number before checking for isNan?

Reply:
IsNan from my understanding works correctly for strings that just have numbers: https://www.w3schools.com/jsref/jsref_isnan.asp

My concern with casting first is that the cast will fail and throw an error before the isNan check. I might be misunderstanding what you’re asking for though.

Reply:
I see, but don't worry, casting an invalid value won't throw any error but just convert the input to a NaN value which will then get caught by the isNaN check